src/components/SMARTBox/PatientBox.js

```diff
@@ -47,17 +46,19 @@ export default class PatientBox extends Component {
     this.getResponses();
   }
 
-  getCoding(request) {
+  getCoding(resource) {
     let code = null;
-    if (request.resourceType === 'DeviceRequest') {
-      code = request?.codeCodeableConcept.coding[0];
-    } else if (request.resourceType === 'ServiceRequest') {
-      code = request?.code?.coding[0];
+    if (resource.resourceType === 'DeviceRequest') {
+      code = resource?.codeCodeableConcept.coding[0];
     } else if (
-      request.resourceType === 'MedicationRequest' ||
-      request.resourceType === 'MedicationDispense'
+      resource.resourceType === 'ServiceRequest' ||
+      resource.resourceType === 'Medication'
     ) {
-      code = request?.medicationCodeableConcept?.coding[0];
+      code = resource?.code?.coding[0];
+    } else if (resource.resourceType === 'MedicationRequest') {
+      code = getDrugCodeFromMedicationRequest(resource);
+    } else if (resource.resourceType === 'MedicationDispense') {
+      code = resource?.medicationCodeableConcept?.coding[0];
```

Comment:
Just leaving this here as a comment that we may need to go back at some point and do the same for medicationDispense